oeps/oep-0030-arch-pii-markup-and-auditing.rst

```diff
@@ -0,0 +1,269 @@
+
+Responsibility for Third-party Service Integrations
+---------------------------------------------------
+When dealing with third-parties that may store PII (ex. Optimizely, Google Analytics, Sailthru) the implementing developer(s) or team members should work with the drivers of the feature and legal counsel to ensure that:
+
+- The third party has a legitimate need for that information to provide the necessary service
+- We send only the minimum necessary information to meet the goals of the feature
+- The third party has an automated, usable way to request that they forget individual learner data (or has a retention policy that results in routine purging of such data)
+- The retirement process is updated to include the third party's forget API before the feature is launched
+
+Github Pull Request Templates
+-----------------------------
+In order to assist developers in remembering to check all new data for PII, each Open edX repository that might store such data will have a GitHub pull request template that reminds the developer and reviewers to check for the addition of such data in their commits and asks them to affirmatively state that either no such data exists or that it does exist and that appropriate retirement steps are, or will be, ready to retire that data before the request is merged.
+
+Docstring Annotations
+---------------------
+**Python**
+
+When adding or modifying **any** data storing models (ex. Django model, MongoDB model) an annotation must be added stating whether the model does or does not store PII. The annotation should be added to the comments of the storage class where the data storage is defined, or the calling method / function if there is no storage class. Calls to third-party services that store data must be annotated to indicate the presence of PII.
+
+It is important to note that under this OEP all Django model classes must be annotated with an assertion of PII / no PII to enable enforcement (see `Enforcement Tooling`_).
+
+These annotation should take the form of a Sphinx-style docstring:
+
+``.. pii:: <description of the PII>. Retired: <location / process of retirement>``
+
+Or, if there is no PII:
+
+``.. pii:: No PII``
+
+Example 1::
+
+    class ApiAccessRequest(TimeStampedModel):
+        """
+        Model to track API access for a user.
+
+        .. pii:: Contains website and employer information about a linked User. Retired: Directly in the LMSAccountRetirementView endpoint.
+        """
+
+If a project requires another project that stores PII, such as Django being used in edx-platform, the developer must annotate the place(s) in code where that package is being called to store the PII with the same docstring annotation as if it were a storage class.
+
+Example 2::
+
+    # ..pii:: Learner email is sent to Segment in the following line and will be associated with analytics data. Retired: In Segment retirement step in Tubular.
+
```

Comment:
Although intentionally not covered in this OEP, is the retirement process/tooling (including Tubular) intended to be available for Open edX installations?

Reply:
Yes. Tubular is a public repo, so we provide all of the pieces we use for retirement except for secrets and our unique configuration. We have some documentation for that process, but also have more tickets to flesh that out with more detail in the backlog.